src/GraphRequest.ts

```diff
@@ -501,19 +494,29 @@ export class GraphRequest {
 	 * @returns The same GraphRequest instance that is being called with
 	 */
 	public query(queryDictionaryOrString: string | KeyValuePairObjectStringNumber): GraphRequest {
-		const otherURLQueryParams = this.urlComponents.otherURLQueryParams;
+		let paramKey: string;
+		let paramValue: string | number;
 		if (typeof queryDictionaryOrString === "string") {
-			const querySplit = queryDictionaryOrString.split("=");
-			const queryKey = querySplit[0];
-			const queryValue = querySplit[1];
-			otherURLQueryParams[queryKey] = queryValue;
+			/* The query key-value pair must be split on the first equals sign to avoid errors in parsing nested query parameters.
+			 Example-> "/me?$expand=home($select=city)" */
+			const indexOfFirstEquals = queryDictionaryOrString.indexOf("=");
```

Comment:
Sorry, I didn't catch this in the first iteration, but we need further validation of queryDictionaryOrString. 

If the string is `test` without any equals sign, indexOfFirstEquals becomes `-1`, `paramKey` becomes empty string, `paramValue` becomes `test`.

Reply:
Handled both the cases mentioned above. 
1. Checking if the first character is ? and cropping it off.
2. If the query parameter is not a valid string or a not a valid parameter appending it the query parameter string. 
Example -> client.api(/me).query("?test").query(123) will be built as "/me?test&123"/. The idea is to not change or replace any user input (even if invalid) and forward the request to the API.